amr-wind/turbulence/RANS/OneEqRANS.cpp

```diff
@@ -0,0 +1,271 @@
+                    const int blank_terrain =
+                        (has_terrain_height) ? 1 - blank_arr(i, j, k) : 1;
+                    mu_arr(i, j, k) =
+                        (std::abs(x3 - probhi[2]) <= 200)
+                            ? 1e-10
+                            : rho_arr(i, j, k) * Cmu_Rt * tlscale_arr(i, j, k) *
+                                  std::sqrt(tke_arr(i, j, k)) * blank_terrain;
+                    const amrex::Real Cmu_prime_Rt = 0.556 / (1 + 0.277 * Rt);
+                    const amrex::Real muPrime =
+                        (std::abs(x3 - probhi[2]) <= 200)
+                            ? 1e-10
+                            : rho_arr(i, j, k) * Cmu_prime_Rt *
+                                  tlscale_arr(i, j, k) *
+                                  std::sqrt(tke_arr(i, j, k)) * blank_terrain;
+                    buoy_prod_arr(i, j, k) = -muPrime * stratification;
+                    shear_prod_arr(i, j, k) *=
+                        shear_prod_arr(i, j, k) * mu_arr(i, j, k);
+                });
+        }
+    }
+
+    mu_turb.fillpatch(this->m_sim.time().current_time());
+}
+
+template <typename Transport>
+void OneEqRANS<Transport>::update_alphaeff(Field& alphaeff)
+{
+
+    BL_PROFILE("amr-wind::" + this->identifier() + "::update_alphaeff");
+    auto lam_alpha = (this->m_transport).alpha();
+    auto& mu_turb = this->m_mu_turb;
+    auto& repo = mu_turb.repo();
+    auto gradT = (this->m_sim.repo()).create_scratch_field(3, 0);
+    fvm::gradient(*gradT, m_temperature);
+    const amrex::GpuArray<amrex::Real, AMREX_SPACEDIM> gravity{
+        m_gravity[0], m_gravity[1], m_gravity[2]};
+    const amrex::Real beta = 1.0 / m_ref_theta;
+    const amrex::Real Cmu = m_Cmu;
+    const int nlevels = repo.num_active_levels();
+    for (int lev = 0; lev < nlevels; ++lev) {
+        for (amrex::MFIter mfi(mu_turb(lev)); mfi.isValid(); ++mfi) {
+            const auto& bx = mfi.tilebox();
+            const auto& muturb_arr = mu_turb(lev).array(mfi);
+            const auto& alphaeff_arr = alphaeff(lev).array(mfi);
+            const auto& lam_diff_arr = (*lam_alpha)(lev).array(mfi);
+            const auto& tke_arr = (*this->m_tke)(lev).array(mfi);
+            const auto& gradT_arr = (*gradT)(lev).array(mfi);
+            const auto& tlscale_arr = (this->m_turb_lscale)(lev).array(mfi);
+            amrex::ParallelFor(
+                bx, [=] AMREX_GPU_DEVICE(int i, int j, int k) noexcept {
+                    amrex::Real stratification =
+                        -(gradT_arr(i, j, k, 0) * gravity[0] +
+                          gradT_arr(i, j, k, 1) * gravity[1] +
+                          gradT_arr(i, j, k, 2) * gravity[2]) *
+                        beta;
+                    amrex::Real epsilon = std::pow(Cmu, 3) *
+                                          std::pow(tke_arr(i, j, k), 1.5) /
+                                          tlscale_arr(i, j, k);
```

Comment:
here you don't have that + 1e-3. Maybe instead of adjusting it everywhere, and forgetting sometimes, it is worth just having a clip step on tlscale at some point so you know it is never going to be 0?